accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/persistence/jpa/monotonic/service/ContiguousIdBlockService.java

```diff
@@ -31,12 +32,12 @@
 
     private ContiguousIdBlockRepository repository;
 
-    private HashMap<String, String> categoryInitValues;
+    private HashMap<String, HashMap<String, Object>> categoryBlockInitializations;
 
-    public ContiguousIdBlockService(ContiguousIdBlockRepository repository, HashMap<String, String>
-            categoryInitValues) {
+    public ContiguousIdBlockService(ContiguousIdBlockRepository repository, HashMap<String, HashMap<String, Object>>
+            categoryBlockInitializations) {
         this.repository = repository;
-        this.categoryInitValues = categoryInitValues;
+        this.categoryBlockInitializations = categoryBlockInitializations;
```

Comment:
If you make the checking here instead then it is enforced at construction and it is not needed to call manually from other classes.

Reply:
we are not aware of the categoryId gonna be used at this stage.